stgit/commands/squash.py

```diff
@@ -86,7 +87,7 @@ def _squash_patches(trans, patches, msg, save_template, no_verify=False):
                 pn,
                 num,
             )
-            msg += "\n%s\n" % trans.patches[pn].data.message_str
+            msg += "\n%s\n\n" % trans.patches[pn].data.message_str.strip()
```

Comment:
Have you considered whether `rstrip()` might be more appropriate than `strip()`? I.e. Should we respect any leading whitespace in a user's commit message?

A concrete example:

```
    Indented message header line

Rest of message.
```

`strip()` and `rstrip()` are going to give different outcomes for the above case. My general sentiment is that it's safe and commendable to normalize the trailing newlines of commit messages, but messing with other user-provided whitespace risks breaking someone's workflow.

Reply:
Totally agree, will fix